include/science/synapse/device.h

```diff
@@ -66,6 +74,32 @@ class Device : public IDevice {
    */
   [[nodiscard]] auto uri() const -> const std::string&;
 
+  /**
+   * Get the logs from the device.
+   * 
+   * @param log_level Minimum log level to retrieve (default: INFO)
+   * @param since_ms Get logs from this many milliseconds ago (optional)
+   * @param start_time_ns Start time in nanoseconds since epoch (optional)
+   * @param end_time_ns End time in nanoseconds since epoch (optional)
+   * @return std::vector<std::string> containing log entries
+   */
+  [[nodiscard]] auto get_logs(
```

Comment:
For both methods, could you return a `science::Status` -- its our convention for reporting any errors (rather than throwing)

You could either return a tuple [Status, string], or, more c++-like, add arguments at the end of each function's signature for outputs, e.g.
`get_logs(const std::string& log_level, ..., std::vector<std::string>* output)`